test/helpers/kubectl.go

```diff
@@ -2220,15 +2220,37 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
 	}
 
 	if !RunsWithKubeProxy() {
-		nodeIP, err := kub.GetNodeIPByLabel(K8s1, false)
-		if err != nil {
-			return fmt.Errorf("Cannot retrieve Node IP for k8s1: %s", err)
+		var k8sServiceHost, k8sServicePort string
+		if GetCurrentIntegration() == CIIntegrationGKE {
+			// On GKE, kube-apiserver is exposed as an external service, but we
```

Comment:
Maybe we should just avoid setting `k8sService{Host,Port}` all together on GKE? The kube-apiserver ClusterIP + port should be already set in the cilium-agent pod via env vars, meaning that cilium-agent should be able to access kube-apiserver.

Reply:
:facepalm: Makes sense. I updated the branch.